R/new.R

```diff
@@ -77,23 +81,37 @@ update_tibble_attrs <- function(x, ...) {
 
 guess_nrow <- function(x) {
   if (!is.null(.row_names_info(x, 0L))) {
-    .row_names_info(x, 2L)
+    list(nrow = .row_names_info(x, 2L), method = "row.names attribute")
   } else if (length(x) == 0) {
-    0L
+    list(nrow = 0L, method = "detected empty list")
   } else {
-    NROW(x[[1L]])
+    list(nrow = NROW(x[[1L]]), method = "length of first column")
   }
 }
 
-validate_nrow <- function(x) {
+validate_nrow <- function(x, nrow_set_method) {
   # Validate column lengths, don't recycle
   lengths <- map_int(x, NROW)
-  first <- .row_names_info(x, 2L)
-
-  bad_len <- lengths != first
+  expected_nrow <- .row_names_info(x, 2L)
+  bad_len <- lengths != expected_nrow
   if (any(bad_len)) {
-    invalid_df_msg(
-      paste0("must be length ", first, ", not "), x, bad_len, lengths[bad_len]
+    vars <- names(x)[bad_len]
+    vars_len <- lengths[bad_len]
+    msg <- paste0("* Column ", tick(vars), " has length ", vars_len, "\n")
+    if (length(bad_len) > 5) {
```

Comment:
`bad_len` looks like a logical vector at this point. Do you need `which()` above (and `length(...) != 0` instead of `any(...)`)? Or perhaps take the length of `vars` (also below)?

Reply:
Yes `bad_len` is a logical vector, and using `length(bad_len)` is a bug! I now use `sum(bad_len)` instead.